staging/src/k8s.io/api/batch/v1/types.go

```diff
@@ -126,12 +126,22 @@ type JobSpec struct {
 	// TTLAfterFinished feature.
 	// +optional
 	TTLSecondsAfterFinished *int32 `json:"ttlSecondsAfterFinished,omitempty" protobuf:"varint,8,opt,name=ttlSecondsAfterFinished"`
+
+	// Suspend specifies whether the Job controller should create Pods or not. If
+	// a Job is created with suspend set to true, no Pods are created by the Job
+	// controller. If a Job is suspended after creation (i.e. the flag goes from
+	// false to true), the Job controller will delete all active Pods associated
+	// with this Job. Users must design their workload to gracefully handle this.
+	// This is an alpha field and requires the SuspendJob feature gate to be
+	// enabled; otherwise this field may not be set to true. Defaults to false.
+	// +optional
+	Suspend bool `json:"suspend,omitempty" protobuf:"varint,10,opt,name=suspend"`
 }
 
 // JobStatus represents the current state of a Job.
 type JobStatus struct {
-	// The latest available observations of an object's current state.
-	// When a job fails, one of the conditions will have type == "Failed".
+	// The latest available observations of an object's current state. Will be
+	// one of following types: "Suspended", "Failed", or "Complete".
```

Comment:
No! It's an open ended list. We don't know what conditions other actors might add, nor do we know what future job controllers might say. It's critical that clients not take a hard dependency on this list never growing.

Reply:
Done.